File: core/src/main/java/xyz/qalcyo/requisite/core/cosmetics/ICosmetic.java
package xyz.qalcyo.requisite.core.cosmetics;

public interface ICosmetic<T> {
    void render(T player, float partialTicks);
    default void render(CosmeticManager<T> cosmeticManager, T player, float partialTicks) {
        if (cosmeticManager.hasCosmetic(player, this)) {
            render(player, partialTicks);
        }
    }
    CosmeticData data();
}

File: core/src/main/kotlin/xyz/qalcyo/requisite/core/cosmetics/ICosmetic.kt
package xyz.qalcyo.requisite.core.cosmetics

interface ICosmetic<T> {
    fun render(player: T, partialTicks: Float)
    fun render(cosmeticManager: CosmeticManager<T>, player: T, partialTicks: Float) {
        if (cosmeticManager.hasCosmetic(player, this)) {
            render(player, partialTicks)
        }
    }

    fun data(): CosmeticData
}